service/matching/ack_manager.go

```diff
@@ -144,7 +146,10 @@ func (m *ackManager) completeTask(taskID int64) int64 {
 		}
 		m.ackLevel = min.(int64)
 		m.outstandingTasks.Remove(min)
+		// decrement backlog as each task gets acked
```

Comment:
was there a specific reason to do the decrement one at a time, instead of say counting how many we want to do and doing them at once? updateApproximateBacklogCount does a lot of locking/atomic ops so it might be nice to consolidate

Reply:
I think the main reason was to be as consistent to the real value of the backlog as possible. If we were to not do it one-by-one, I assume we would have to regularly update the counter for a batch of completed tasks which could further increase the error gap between the actual and the approximate value right?

Reply:
I'm just talking about doing the call once before this function returns vs several times in this loop. we're not doing any IO or blocking here, we're just updating an in-memory data structure.